### PR TITLE
8th commit:

### PR DESCRIPTION
**Pull request by waspp01**
- Added ShoppingCart Entity, Service, Repository, DTOs , Controller, Mapper
- Added CartItem entity, DTOs, Service, Repository, Mapper
- Added ShoppingCart creation to UserServiceImpl
- Added required liquibase changes

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/controller/ShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping cart management", description = "Endpoints for managing shopping cart")
+@RestController
+@RequestMapping(value = "/cart")
```
Comment (koval-kn) [inline src/main/java/com/example/onlinebookstore/controller/ShoppingCartController.java]:
```suggestion
@RequestMapping("/cart")
```

File: src/main/java/com/example/onlinebookstore/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public ShoppingCartDto addCartItem(Long userId,
+                                   CreateCartItemRequestDto requestDto) {
+        ShoppingCart shoppingCart = getShoppingCartById(userId);
+        List<CartItem> cartItems = shoppingCart.getCartItems();
```
Comment (koval-kn) [inline src/main/java/com/example/onlinebookstore/service/impl/ShoppingCartServiceImpl.java]: consider extracting this check to the separate method

**Review by koval-kn — CHANGES_REQUESTED** [review]
consider to improve your commit naming in future

**Comment by waspp01** [comment]
> consider to improve your commit naming in future

What do you mean? Should I add some info about implemented things instead of just number of commit?

**Comment by waspp01** [comment]
> consider to improve your commit naming in future

What do you mean? Should I add some info about implemented things instead of just number of commit?

**Comment by koval-kn** [comment]
> > consider to improve your commit naming in future
> 
> What do you mean? Should I add some info about implemented things instead of just number of commit?

yeah :)

**Review by koval-kn — APPROVED** [review]

**Pull request merged**